main.js

```diff
@@ -620,6 +662,256 @@ app.post('/delete-admin', async function(req, res) {
+	}
+	const itemProps = JSON.parse(req.body.itemProps);
+	const newItem = {
+		name: itemProps.name,
+		quantity: itemProps.quantity,
+		description: itemProps.description,
+		image: {
+			data: {
+				$binary: {
+					base64: req.file.buffer
+				}
+			},
+			contentType: req.file.mimetype
+		},
+		completed: itemProps.completed,
+		required: itemProps.required,
+		_id: new mdb.ObjectId()
+	}
+	try {
+		const result = await db.collection('BBY-6_users').updateOne(filterQuery, {$push: {"kits.$.kit": newItem}});
+		if(!result) {
+			// Could not find user 
+			res.status(500).send("couldNotFindUser");
+			return;
+		} else if (result.matchedCount === 0) {
+			// Could not find kit
+			res.status(404).send("couldNotFindKit");
+			return;
+		}
+		res.status(200).send("itemAddedSuccessfully");
+	} catch(e) {
+		res.status(500).send("serverIssue");
+	}
+})
+
+app.patch('/edit-item', async (req,res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	if(!req.body || !req.body._id || !req.body.itemProps) {
+		res.status(400).send("missingBodyArguments");
+		return;
+	}
+	const filterQuery = {
+		'_id': new mdb.ObjectId(req.session._id),
+		'kits._id': new mdb.ObjectId(req.body._id),
+	}
+	try {
+		req.body.itemProps._id = new mdb.ObjectId(req.body.itemProps._id);
+		const result = await db.collection('BBY-6_users').updateOne(filterQuery, {$set: {"kits.$[i].kit.$[j]": req.body.itemProps}}, 
+			{
+				arrayFilters: [ 
+					{"i._id": new mdb.ObjectId(req.body._id)}, 
+					{"j._id": new mdb.ObjectId(req.body.itemProps._id)}
+				]
+			}
+		);
+		if(!result) {
+			// Could not find user 
+			res.status(500).send("couldNotFindUser");
+			return;
+		} else if (result.matchedCount === 0) {
+			// Could not find kit
+			res.status(404).send("couldNotFindKit");
+			return;
+		}
+		res.status(200).send("kitUpdatedSuccessfully");
+	} catch(e) {
+		console.log(e)
+		res.status(500).send("serverIssue");
+	}
+})
+
+app.delete('/delete-item', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	if(!req.body || !req.body._id || !req.body.itemId) {
+		res.status(400).send("missingBodyArguments");
+		return;
+	}
+	const filterQuery = {
+		'_id': new mdb.ObjectId(req.session._id),
+		'kits._id': new mdb.ObjectId(req.body._id)
+	}
+	try {
+		req.body._id = new mdb.ObjectId(req.body._id);
+		const result = await db.collection('BBY-6_users').updateOne(filterQuery, 
+			{
+				$pull: {
+						"kits.$[i].kit": {
+							"_id": new mdb.ObjectId(req.body.itemId),
+							"required": false
+						}
+				}
+			},
+			{
+				arrayFilters: [ 
+					{"i._id": new mdb.ObjectId(req.body._id)}
+				]
+			}
+		);
+		if(!result) {
+			// Could not find user 
+			res.status(500).send("couldNotFindUser");
+			return;
+		} else if (result.matchedCount === 0) {
+			// Could not find kit
+			res.status(404).send("couldNotFindKit");
+			return;
+		} else if (result.modifiedCount === 0) {
+			res.status(401).send("cannotDeleteRequiredItem");
+			return;
```

Comment:
Not immediately obvious that the database query won't hit required items – maybe leave a comment up at the query?